docs/API-DESIGN.md

```diff
@@ -0,0 +1,881 @@
+# Reaper Go Client - API Enhancement Proposal
+
+
+# State of the Art
+
+The full REST API is documented here: [http://cassandra-reaper.io/docs/api](http://cassandra-reaper.io/docs/api). 
+
+However this page is not entirely up to date. A more exhaustive list is as follows; implemented methods are shown in bold (even if only partially implemented):
+
+
+
+*   Ping
+    *   <b>`GET /ping`</b>
+    *   `HEAD /ping`
+*   Authentication
+    *   `POST /login`
+    *   `POST /logout`
+    *   `GET /jwt`
+*   Cluster
+    *   <b>`GET /cluster`</b>
+    *   <b>`POST /cluster`</b>
+    *   `POST /cluster/auth`
+    *   <b>`DELETE /cluster/{cluster_name}`</b>
+    *   <b>`GET /cluster/{cluster_name}`</b>
+    *   <b>`PUT /cluster/{cluster_name}`</b>
+    *   `PUT /cluster/{cluster_name}/auth`
+    *   `GET /cluster/{cluster_name}/tables`
+*   Repair Runs
+    *   `GET /repair_run`
+    *   `POST /repair_run`
+    *   `GET /repair_run/cluster/{cluster_name}`
+    *   `POST /repair_run/purge`
+    *   `DELETE /repair_run/{id}`
+    *   `GET /repair_run/{id}`
+    *   `PUT /repair_run/{id}`
+    *   `PUT /repair_run/{id}/intensity/{intensity}`
+    *   `GET /repair_run/{id}/segments`
+    *   `POST /repair_run/{id}/segments/abort/{segment_id}`
+    *   `PUT /repair_run/{id}/state/{state}`
+*   Repair Schedules
+    *   <b>`GET /repair_schedule`</b>
+    *   `POST /repair_schedule`
+    *   <b>`GET /repair_schedule/cluster/{cluster_name}`</b>
+    *   `POST /repair_schedule/start/{id}`
+    *   `DELETE /repair_schedule/{id}`
+    *   `GET /repair_schedule/{id}`
+    *   `PUT /repair_schedule/{id}`
+    *   `GET /repair_schedule/{clusterName}/{id}/percent_repaired`
+*   Snapshot
+    *   `GET /snapshot/cluster/{clusterName}`
+    *   `POST /snapshot/cluster/{clusterName}`
+    *   `DELETE /snapshot/cluster/{clusterName}/{snapshotName}`
+    *   `GET /snapshot/{clusterName}/{host}`
+    *   `POST /snapshot/{clusterName}/{host}`
+    *   `DELETE /snapshot/{clusterName}/{host}/{snapshotName}`
+*   Node
+    *   `GET /node/clientRequestLatencies/{clusterName}/{host}`
+    *   `GET /node/compactions/{clusterName}/{host}`
+    *   `GET /node/dropped/{clusterName}/{host}`
+    *   `GET /node/streams/{clusterName}/{host}`
+    *   `GET /node/tokens/{clusterName}/{host}`
+    *   `GET /node/tpstats/{clusterName}/{host}`
+*   Diagnostic Events
+    *   `GET /diag_event/sse_listen/{id}`
+    *   `GET /diag_event/subscription`
+    *   `POST /diag_event/subscription`
+    *   `DELETE /diag_event/subscription/{id}`
+    *   `GET /diag_event/subscription/{id}`
+
```

Comment:
I may have missed it, but is here is a place for this in the original docs.  

`GET /crypto/encrypt/{text}`

Does this one go away or will it be modified as part of this PR?

Reply:
Nice catch again! Indeed there is a `CryptoResource` in Reaper. I'm not sure though if it would make sense to expose it in the Reaper client. @adejanovski what's your opinion?

Reply:
Added the Crypto resource in b4c3654. We'll see later if it's useful from a client perspective.